authentication/auth.go

```diff
@@ -0,0 +1,162 @@
+package authentication
+
+import (
+	"fmt"
+	"log"
+	"time"
+	"errors"
+	"regexp"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/satori/go.uuid"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/collisdigital/go-launch-a-survey/settings"
+)
+
+func loadEncryptionKey() (key *rsa.PublicKey, error error) {
+
+	encryptionKeyPath := settings.GetSetting("JWT_ENCRYPTION_KEY_PATH")
+	if keyData, err := ioutil.ReadFile(encryptionKeyPath); err == nil {
```

Comment:
would be more idiomatic to handle error conditions first, i.e. `err != nil` - can make capturing the non-error return value tricker though

i'd probably name the error var `err` instead of `error` to avoid package/type name conflicts (generally works ok, but can be confusing)